packages/docs/src/index.tsx

```diff
@@ -29,10 +36,45 @@ const versions = require<string[]>("./generated/versions.json")
         url: `https://palantir.github.io/blueprint/docs/${version}`,
         version,
     } as IPackageInfo));
-
-const propsStore = new PropsStore(require<IInterfaceEntry[]>("./generated/props.json"));
 /* tslint:enable:no-var-requires */
 
+const propsStore = new PropsStore(docs.ts);
+function resolveInterface(name: string, key: React.Key) {
+    const props = propsStore.getProps(name);
+    return <PropsTable key={key} name={name} props={props} />;
+}
+
+const Heading: React.SFC<{ depth: number, header: string, reference: string }> =
+    ({ depth, header, reference }) => (
+        // use createElement so we can dynamically choose tag based on depth
+        React.createElement(`h${depth}`, { className: "kss-title" },
+            <a className="docs-anchor" key="anchor" name={reference} />,
+            <a className="docs-anchor-link" href={"#" + reference} key="link">
+                <span className="pt-icon-standard pt-icon-link" />
+            </a>,
+            header,
+        )
+    );
+
+function renderHeading(depth: number) {
+    return (heading: string, key: React.Key, page: IPageData): JSX.Element => {
+        const ref = (depth === 1 ? page.reference : slugify(page.reference, heading));
+        return <Heading depth={depth} header={heading} key={key} reference={ref} />;
+    };
+}
+
+// tslint:disable:object-literal-key-quotes
+const TAGS = {
+    "#": renderHeading(1),
+    "##": renderHeading(2),
+    "###": renderHeading(3),
+    "####": renderHeading(4),
+    interface: resolveInterface,
+    page: () => undefined as JSX.Element,
```

Comment:
every tag must have a handler, but pages don't get rendered in the flow.

i'd like to eventually render a cool link, kinda like the Resources page.